dune

```diff
@@ -0,0 +1,6 @@
+(dirs :standard \ generated_tests)
+
+(env
+ (dev
+  (flags
+   (:standard -warn-error -A))))
```

Comment:
Why are you using `-warn-error -A` here? If we only want to disable one or two specific warnings, we can just remove them (see `ocamlc -warn-help` for the list of warnings with their numbers). Or maybe you decided to just never have warnings turned into errors? (Sometimes the errors makes the build too fragile, but sometimes it's the right choice to force us to fix the code, so I'm not sure what I think should be the default here, but I have the feeling that this may be a bit too broad.)

Reply:
Yes, my first intention was to make all warnings non-fatal, but working for the second PR, I came to appreciate fatal warnings. I intend to make only the deprecated alerts non-fatal.

Reply:
Thanks! I think it would be good.